eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala

```diff
@@ -103,13 +116,15 @@ class BitcoinCoreWallet(rpcClient: BitcoinJsonRPCClient)(implicit ec: ExecutionC
     } yield MakeFundingTxResponse(fundingTx, outputIndex, fee)
   }
 
-  override def commit(tx: Transaction): Future[Boolean] = publishTransaction(tx)
-    .map(_ => true) // if bitcoind says OK, then we consider the tx successfully published
-    .recoverWith { case JsonRPCError(e) =>
+  override def commit(tx: Transaction): Future[Boolean] = publishTransaction(tx).transformWith {
```

Comment:
`transformWith` looks better than what we had before, but I would have made a minimal fix for this sensitive piece of code. Maybe all things considered it's better to make it more readable. We really really do not want to return `false` if there is the slightest chance that the tx has been published.

Reply:
We return `false` only if `publish` failed AND the tx is not in the mempool, as before, but in that case we also rollback its inputs. I tried to minimize the diff here but it was ugly.